data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -424,16 +426,25 @@ private void processRecord(final AcknowledgementSet acknowledgementSet, final Re
                 bufferAccumulator.add(record);
```

Comment:
what happens to this record that was caught in exception ? Is this lost ?

Reply:
I see, the consumer retries in while loop.

Reply:
No. The record is put in buffer eventually because we are in an infinite loop here.